R/sendQuery.R

```diff
@@ -231,3 +233,19 @@ formatWarnings <- function(dat) {
   dat <- paste(dat, collapse = "\n")
   dat
 }
+
+fixTimezone <- function(dat, tz) {
+  timeFields <- unlist(lapply(dat, inherits, what = "POSIXct"))
+  if (length(timeFields) == 0) return(dat)
+
+  lapply(names(dat)[timeFields], function(n) {
+    dat[[n]] <<- forceTZ(dat[[n]], tz = tz)
+  })
+  dat
+}
+
+forceTZ <- function(x, tz) {
+  attr(x, "tzone") <- "UTC"
+  x <- as.character(x)
```

Comment:
verliert `x` nicht bei der Umwandlung das Attribut 'tzone'? Bzw. was passiert wenn wir das Attribut explizit setzen?

Reply:
Das Ziel ist auch das Attribut loszuwerden. Wenn ich das Attribut explizit setze verändert sich auch der Timestamp:

```r
> now <- Sys.time()
> now
[1] "2020-02-07 09:42:19 CET"
> attributes(now)
$class
[1] "POSIXct" "POSIXt" 

> attr(now, "tzone") <- "UTC"
> now
[1] "2020-02-07 08:42:19 UTC"
```

Es gibt im lubridate Paket auch eine entsprechende Funktion, die vermutlich performanter ist, da sie in C implementiert ist. Aber lubridate ist ne dicke Abhängigkeit:

```r
> now <- Sys.time()
> now
[1] "2020-02-07 09:44:28 CET"
> lubridate::force_tz(now, "UTC")
[1] "2020-02-07 09:44:28 UTC"
```

Reply:
Timings
```
> system.time(lapply(rep(now, 10000), lubridate::force_tz, tzone = "UTC"))
   user  system elapsed 
  0.062   0.000   0.062 

> system.time(lapply(rep(now, 10000), forceTZ, tz = "UTC"))
   user  system elapsed 
  0.714   0.068   0.783 
```

Reply:
Verstehe